chatbot-core/backend/app/models/chat.py

```diff
@@ -0,0 +1,255 @@
+    chat_session_id: Mapped[UNIQUEIDENTIFIER] = mapped_column(
+        ForeignKey("chat_sessions.id", ondelete="CASCADE"), nullable=False
+    )
+    alternate_agent_id: Mapped[Optional[UNIQUEIDENTIFIER]] = mapped_column(ForeignKey("agents.id"), nullable=True)
+    parent_message_id: Mapped[Optional[UNIQUEIDENTIFIER]] = mapped_column(ForeignKey(CHAT_MESSAGES_ID), nullable=True)
+    child_message_id: Mapped[Optional[UNIQUEIDENTIFIER]] = mapped_column(ForeignKey(CHAT_MESSAGES_ID), nullable=True)
+    message: Mapped[str] = mapped_column(Text)
+    prompt_id: Mapped[Optional[UNIQUEIDENTIFIER]] = mapped_column(ForeignKey("prompts.id"), nullable=True)
+    token_count: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
+    message_type: Mapped[ChatMessageType] = mapped_column(
+        SQLAlchemyEnum(ChatMessageType, native_enum=False), nullable=False
+    )
+    error_type: Mapped[Optional[ChatMessageErrorType]] = mapped_column(
+        SQLAlchemyEnum(ChatMessageErrorType), nullable=True
+    )
+    error: Mapped[Optional[str]] = mapped_column(String(500), nullable=True)
+    created_at: Mapped[datetime] = mapped_column(
+        DateTime(timezone=True), server_default=func.now(), default=lambda: datetime.now(timezone.utc)
+    )
+    updated_at: Mapped[datetime] = mapped_column(
+        DateTime(timezone=True), server_default=func.now(), onupdate=lambda: datetime.now(timezone.utc)
+    )
+    deleted_at: Mapped[Optional[datetime]] = mapped_column(DateTime(timezone=True), nullable=True, default=None)
+
+    # Define relationships
+    user: Mapped["User"] = relationship("User", back_populates="chat_messages")
+    chat_session: Mapped["ChatSession"] = relationship("ChatSession", back_populates="chat_messages")
+    agent: Mapped[Optional["Agent"]] = relationship("Agent", back_populates="chat_messages")
+    prompt: Mapped[Optional["Prompt"]] = relationship("Prompt", back_populates="chat_messages")
+
+    @validates("token_count")
+    def validate_token_count(self, key, token_count) -> Union[int, None]:
+        """
+        Validate token count.
+
+        Args:
+            key (str): Key.
+            token_count (int): Token count.
+        """
+        if token_count < 0:
+            raise ValueError("Token count must be greater than or equal to zero.")
+        return token_count
+
+
+class ChatMessageRequest(BaseModel):
+    """
+    Pydantic model for creating a new chat message.
+    Provides validation for incoming chat message requests.
+    """
+
+    id: Optional[UUID] = Field(None, description="Chat message id")
+    alternate_agent_id: Optional[UUID] = Field(None, description="Alternate agent id")
+    parent_message_id: Optional[UUID] = Field(None, description="Parent message id")
+    child_message_id: Optional[UUID] = Field(None, description="Child message id")
+    message: Optional[str] = Field(None, description="Message text", min_length=1, max_length=10000)
```

Comment:
Need another validation for input sanitization checking. This will help prevent:
- XSS attacks
- Control character injection
- Overly long messages
- Leading/trailing whitespace issues

And I think it's neccessary to implement.
```python
@validates("message")
def sanitize_chat_message(self, message: str) -> str:
    # Remove any control characters
    message = ''.join(char for char in message if ord(char) >= 32)
    
    # Limit message length (example: 2000 characters)
    message = message[:2000]
    
    # Strip whitespace
    message = message.strip()
    
    # Optional: Remove or replace HTML/special characters
    message = (message
        .replace('<', '&lt;')
        .replace('>', '&gt;')
        .replace('"', '&quot;'))
    
    return message
```

Example might not run correctly (I generate using my best friend, double check pls!)

Reply:
I think the best way is to preserve the raw message in our database. We can have further separate transformation on that raw message when we work with LLM or something like this.

I think I will keep the checks `Limit message length` and `Strip whitespace`.

Reply:
Check out this commit a18de5f7165ca854cd119f6c702aacc9a0b04299.

Reply:
Sure, that's fine.